android/src/main/java/com/taplytics/react/TaplyticsReactModule.java

```diff
@@ -243,6 +248,12 @@ public void onNewSession() {
                 params.putBoolean("value", true);
                 sendEvent("newSession", params);
             }
+            @Override
+            public void onError() {
+                WritableMap params = Arguments.createMap();
+                params.putBoolean("value", false);
+                sendEvent("newSession", params);
```

Comment:
Does it make sense to still emit this event on error? What if people are using the listener but ignoring the value?

Reply:
@ajwootto yeah you're right that doesn't make sense.
Would it be better to just use a promise and then use a then and catch in the index.js which also resolves and rejects?

Reply:
I dont think so because this is technically a "listener" which means it can be called many times at any time by the SDK.

Reply:
One thing you could potentially do is change the DeviceEmitter listener function on the JS side to take two parameters, a "newSession" callback and an "onError" callback

Reply:
@ajwootto I see. Just leave it empty then?